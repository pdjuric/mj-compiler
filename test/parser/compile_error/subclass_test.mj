program main_test

class A extends int {}
class C extends B {}
class D extends D {}

{
    void main() {}
}