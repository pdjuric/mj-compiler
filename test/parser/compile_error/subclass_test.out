ERROR 17:11:45,371 - line   3 - int is not a class
ERROR 17:11:45,372 - line   4 - Undeclared type [B].
ERROR 17:11:45,372 - line   4 - Undeclared type [B].
ERROR 17:11:45,372 - line   5 - A class cannot be its own superclass [D].